### ISSUE TITLE
Error while checking syntax automatically: (wrong-type-argument stringp nil) for JS files

### ISSUE THREAD

**Issue by jonathanchu** [flycheck/flycheck#1115]
After updating to the latest version of flycheck on Melpa (flycheck-20160927.1734) I started getting this error when accessing any JS files.

Here's a screenshot of my emacs where I followed [this great article](http://www.lunaryorn.com/2015/11/29/reproduce-bugs-in-emacs-Q.html) from @lunaryorn to initially reproduce the error.

<img width="630" alt="screenshot 2016-09-28 00 07 11" src="https://cloud.githubusercontent.com/assets/66164/18900529/07f1e256-8510-11e6-835c-9ba695718604.png">

I ran `M-x flycheck-verify-setup` and the only thing is that stood out to me is that I don't have eslint installed.  Is this the issue?  I was able to bisect this error to this [commit](https://github.com/flycheck/flycheck/commit/ec31874a662fdf1dcd174bca99af870a20415ceb).  Using a version of flycheck before this commit does not produce this error.

Here are the rest of the details:
* Your operating system
OS X (El Capitan) v10.11.6

* Your Emacs version from `M-x emacs-version`
Emacs 25.1.1

* Your Flycheck version from `M-x flycheck-version`
30snapshot (package: 20160927.1734)

**Comment by cpitclaudel** [comment]
Hi @jonathanchu,

Thanks for the report! And sorry for the trouble.  One more data point: can you run `M-x toggle-debug-on-error` and reproduce the error? Then Emacs will print a stack trace, which will help us debug.

Thanks for your help!

**Comment by lunaryorn** [comment]
@cpitclaudel I think I've found the problem (`:enabled` of eslint uses the executable without checking whether it exists) but I've got a more comprehensive change for `:enabled` in #1116 which also fixes this issue along the way.

@jonathanchu If you could spare a couple of minutes I'd be very happy if you could test #1116 :)

Thank you for reporting this issue, we hope to be able to merge the fix soon.  Sorry for the inconvenience.  As a temporary workaround, disable `javascript-eslint` with `C-c ! x javascript-eslint`.

**Comment by lunaryorn** [comment]
By the way, I'm happy to hear that you liked my blog post :blush:

**Comment by jonathanchu** [comment]
@lunaryorn my pleasure, it's the least I could do if I'm simply reporting an issue with no proposed or attached fix 😉 

I just did a checkout of #1116 locally and installed into my emacs - I can confirm that this _does_ fix this issue.  Thanks so much for the quick fix and all the great work on flycheck! @lunaryorn @cpitclaudel 

:beers:

**Comment by lunaryorn** [comment]
@jonathanchu Great.  We merged the fix, the next MELPA build should do away the issue.

Sorry for the inconvenience, and thank you for your very thorough and detailed report :+1:

**Comment by jonathanchu** [comment]
@lunaryorn awesome, thanks so much for the quick bug fix and great responses.  Thank you ALL! :heart: 😄

**Comment by suvidh** [comment]
I am facing the same error, but I am a noob to emacs. Can someone give me a step by step guide?

**Comment by 4imothy** [comment]
Did you figure out the issue @suvidh? I am trying to work on C files. @cpitclaudel I ran with toggle-debug-on-error and got this:
```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  signal(wrong-type-argument (stringp nil))
  flycheck-buffer()
  flycheck-buffer-automatically()
  flycheck-perform-deferred-syntax-check()
```

Clang attaches correctly because I get this message and also see a correct error counter on the bottom but the error text doesn't show:
```
LSP :: Connected to [clangd:29997 `path`].
```

**Comment by StillerHarpo** [comment]
@4imothy Do you use `doom-modeline`? I had the same problem. Removing `doom-modeline` fixed it. See https://www.reddit.com/r/emacs/comments/15kksbx/doommodeline_seems_to_be_breaking_flycheck/
